dev-client/src/components/sites/LocationDashboardScreen.tsx

```diff
@@ -0,0 +1,89 @@
+import {ScreenDefinition, useNavigation} from '../../screens/AppScaffold';
+import {useDispatch, useSelector} from '../../model/store';
+import {HeaderTitle} from '@react-navigation/elements';
+import {useTranslation} from 'react-i18next';
+import RadioBlock, {IconLabel} from '../common/RadioBlock';
+import {SitePrivacy, updateSite} from 'terraso-client-shared/site/siteSlice';
+import {useCallback} from 'react';
+import {Box, Text, useTheme} from 'native-base';
+import {Icon, IconButton} from '../common/Icons';
+import {ScreenScaffold} from '../../screens/ScreenScaffold';
+import {MainMenuBar} from '../../screens/HeaderIcons';
+
+type Props = {siteId: string};
+
+const LocationDashboardView = ({siteId}: Props) => {
+  const {t} = useTranslation();
+  const {colors} = useTheme();
+  const dispatch = useDispatch();
+  const site = useSelector(state => state.site.sites[siteId]);
+  const project = useSelector(state =>
+    site.projectId === undefined
+      ? undefined
+      : state.project.projects[site.projectId],
+  );
+
+  const onSitePrivacyChanged = useCallback(
+    (privacy: SitePrivacy) => dispatch(updateSite({...site, privacy})),
+    [site, dispatch],
+  );
+
+  return (
+    <ScreenScaffold padding="22">
+      {!project && (
+        <RadioBlock
+          label={
+            <IconLabel
+              label={t('site.dashboard.privacy.site')}
+              icon={<Icon name="info" />}
+            />
+          }
+          options={{
+            PUBLIC: {text: t('privacy.PUBLIC.title')},
+            PRIVATE: {text: t('privacy.PRIVATE.title')},
+          }}
+          blockName="site-privacy"
+          onChange={onSitePrivacyChanged}
+          value={site.privacy}
+          oneLine
+        />
+      )}
+      {project && (
+        <Box background={colors.grey[200]}>
+          <Text>
+            <Text bold>
+              {t('site.dashboard.privacy.project', {
+                privacy: t(`'site.dashboard.privacy.${site.privacy}.inline'`),
+              })}
+            </Text>
+            <Text>{project.name}</Text>
+          </Text>
+        </Box>
+      )}
+    </ScreenScaffold>
+  );
+};
+
+export const LocationDashboardScreen: ScreenDefinition<Props> = {
+  View: LocationDashboardView,
+  options: ({siteId}) => ({
+    headerBackVisible: false,
+    headerLeft: MainMenuBar,
+    headerRight: ({tintColor}) => {
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const {navigate} = useNavigation();
+      return (
+        <IconButton
+          name="settings"
+          _icon={{color: tintColor}}
+          onPress={() => navigate('SITE_SETTINGS', {siteId})}
+        />
+      );
+    },
+    headerTitle: props => {
```

Comment:
note: i realized i got myself in a bit of a twist while doing the screen refactor, so i'm no longer wrapping the `HeaderTitle` arg because it was unnecessary, and we would've had to wrap all the other args (like `headerRight`) in the same way which would've been quite annoying. this change has been propogated everywhere, the actual change is visible in the (now much simpler) `AppScaffold.tsx` code